runtime/serde/serde-benchmarks/jvm/src/aws/smithy/kotlin/serde/benchmarks/json/twitter/model/Entities.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+// Code generated by smithy-kotlin-codegen. DO NOT EDIT!
+
+package aws.smithy.kotlin.serde.benchmarks.json.twitter.model
+
+class Entities private constructor(builder: BuilderImpl) {
```

Comment:
**Comment**: Checking in this generated code means it'll get out of sync with our codegen, thus separating the benchmark results from reality. Can we codegen this as part of the build instead?

Reply:
If I could have done it easily I would have.

* we don't have an easy way to just spit out _only_ the document serializers/deserializers + model types
* there is no protocol other than the AWS ones to choose from which means
    * we would have to define a separate package, implement a dummy json/xml/whatever format protocol. 
    * depend on it from this benchmark build
    * implement logic to have `compileKotlin` tasks run after codegen tasks
    * copy/move the bits we need from codegen into the main sourceSet

It's certainly possible but the amount of time to invest in this didn't seem worth it right now. If we're that worried I'd rather just remove the benchmarks for serialize/deserialize and only add them once we have the time to invest in something like above.

Reply:
Hmm, I see the problems. If we can't keep the types and structurers/destructurers up to date easily, that curbs the value of checking in benchmarks for those things (for me). I now lean towards removing them but my opinion isn't strong enough to make it a requirement.

I bet some refactoring of our codegen could get us a clean way to generate just the layers we need for this test. This also seems like a reason why it may be good for non-AWS protocols to be defined and for **smithy-kotlin** to support them.

Reply:
It would be useful to be able to just generate entities OR entities + "document" serializers/deserializers for some format (json/xml/etc). This wouldn't require a protocol and I could see people finding it useful. 

The gradle problems would be solvable it just takes time to invest in getting the build setup.